src/pages/RealmExplorer.js

```diff
@@ -3,201 +3,214 @@
 
-import { ListView } from 'realm/react-native';
+import { schema } from '../database/schema';
 
-import globalStyles from '../globalStyles';
+const REALM_TYPES = {
+  boolean: ['bool'],
+  date: ['date'],
+  number: ['int', 'float', 'double'],
+  string: ['string'],
+  array: ['list'],
+  object: ['object'],
+};
 
-const OBJECT_TYPES = [
-  'Address',
-  'Item',
-  'ItemBatch',
-  'ItemDepartment',
-  'ItemCategory',
-  'ItemStoreJoin',
-  'Transaction',
-  'TransactionItem',
-  'TransactionBatch',
-  'TransactionCategory',
-  'MasterList',
-  'MasterListItem',
-  'MasterListNameJoin',
-  'Name',
-  'NameStoreJoin',
-  'NumberSequence',
-  'NumberToReuse',
-  'Requisition',
-  'RequisitionItem',
-  'Setting',
-  'SyncOut',
-  'Stocktake',
-  'StocktakeItem',
-  'StocktakeBatch',
-  'User',
-];
+const typeMapper = new Map(
+  Object.entries(REALM_TYPES)
+    .map(([type, realmTypes]) => realmTypes.map(realmType => [realmType, type]))
+    .flat()
+);
 
-/**
- * A page to explore the contents of the local database. Allows searching for any
- * database object type, and will show the related data in a table.
- *
- * @prop   {Realm}                database    App wide database.
- * @state  {ListView.DataSource}  dataSource  DataTable input, used to update rows being rendered.
- * @state  {Realm.Results}        data        Holds the data that gets put into the dataSource.
- */
-export class RealmExplorer extends React.Component {
-  constructor(props) {
-    super(props);
-    const dataSource = new ListView.DataSource({
-      rowHasChanged: (row1, row2) => row1 !== row2,
-    });
-    this.state = {
-      data: null,
-      dataSource,
-    };
-    this.unfilteredData = null;
-    this.onSearchChange = this.onSearchChange.bind(this);
-    this.onFilterChange = this.onFilterChange.bind(this);
-    this.renderHeader = this.renderHeader.bind(this);
-    this.renderRow = this.renderRow.bind(this);
-  }
+const getObjectTypes = ({ schema: objectSchemas }) =>
+  objectSchemas.map(({ schema: objectSchema }) => objectSchema.name);
 
-  componentWillMount() {
-    const { database } = this.props;
-    const { dataSource } = this.state;
+const parseType = realmType => {
+  const { type } = realmType;
+  if (type) return parseType(type);
+  const parsedType = typeMapper.get(realmType);
+  return parsedType;
+};
 
-    const data = database.objects('User');
-    this.setState({
-      data,
-      dataSource: dataSource.cloneWithRows(data),
-    });
+const getObjectFields = ({ schema: objectSchemas }) =>
+  objectSchemas
+    .map(({ schema: objectSchema }) => {
+      const { name, properties } = objectSchema;
+      const fields = Object.entries(properties)
+        .map(([field, type]) => ({
+          [field]: parseType(type),
+        }))
+        .reduce((acc, field) => ({ ...acc, ...field }), {});
+      return { [name]: fields };
+    })
+    .reduce((acc, object) => ({ ...acc, ...object }));
+
+const OBJECT_TYPES = getObjectTypes(schema);
```

Comment:
a little more differentiation between realm object types and primitives would be appreciated took a sec to realize this was the realm object types and not related to the primitive mapping

Reply:
Good point. 589030767bf1963dcbb6c3ef10a11973f2e48ec0.